ffi/diplomat/src/timezone.rs

```diff
@@ -33,143 +22,123 @@ pub mod ffi {
     use icu_timezone::TimeZoneBcp47Id;
     use icu_timezone::ZoneVariant;
 
-    #[diplomat::rust_link(icu::timezone::GmtOffset, Struct)]
-    pub struct ICU4XGmtOffset {
-        pub(super) _private_data: i32,
-    }
-
-    pub struct ICU4XOptionalGmtOffset {
-        pub(super) _private_data: i32,
-    }
-
+    #[diplomat::opaque]
     #[diplomat::rust_link(icu::timezone::CustomTimeZone, Struct)]
-    pub struct ICU4XCustomTimeZone {
-        pub gmt_offset: ICU4XOptionalGmtOffset,
-        pub time_zone_id: ICU4XOptionalTinyAsciiStr8,
-        pub metazone_id: ICU4XOptionalTinyAsciiStr4,
-        pub zone_variant: ICU4XOptionalTinyAsciiStr2,
-    }
+    pub struct ICU4XCustomTimeZone(pub CustomTimeZone);
 
     #[diplomat::opaque]
     #[diplomat::rust_link(icu::timezone::MetaZoneCalculator, Struct)]
     pub struct ICU4XMetaZoneCalculator(pub MetaZoneCalculator);
 
-    impl ICU4XGmtOffset {
-        pub fn create_from_str(s: &str) -> DiplomatResult<ICU4XGmtOffset, ICU4XError> {
-            GmtOffset::from_str(s)
-                .map(Into::into)
+    impl ICU4XCustomTimeZone {
+        pub fn create_from_str(s: &str) -> DiplomatResult<Box<ICU4XCustomTimeZone>, ICU4XError> {
+            CustomTimeZone::from_str(s)
+                .map(ICU4XCustomTimeZone::from)
+                .map(Box::from)
                 .map_err(Into::into)
                 .into()
         }
 
         #[diplomat::rust_link(icu::timezone::GmtOffset::try_from_offset_seconds, FnInStruct)]
-        pub fn try_from_offset_seconds(seconds: i32) -> DiplomatResult<ICU4XGmtOffset, ICU4XError> {
-            GmtOffset::try_from_offset_seconds(seconds)
-                .map(Into::into)
-                .map_err(Into::into)
-                .into()
+        pub fn try_set_gmt_offset_seconds(
```

Comment:
nit: we need a way to set this to None too (`clear_gmt_offset()`)

Reply:
ee373eb (and fixed in a later commit)